sys/contrib/ck/include/gcc/ck_pr.h

```diff
@@ -183,18 +184,22 @@ CK_PR_CAS_S(8,  uint8_t)
 CK_CC_INLINE static bool
 ck_pr_cas_ptr_value(void *target, void *compare, void *set, void *v)
 {
-	set = __sync_val_compare_and_swap((void **)target, compare, set);
-	*(void **)v = set;
-	return (set == compare);
+	bool z;
+	z = __atomic_compare_exchange((void **)target, &compare, &set, false,
+	    __ATOMIC_SEQ_CST, __ATOMIC_SEQ_CST);
+	*(void **)v = compare;
+	return z;
 }
 
 #define CK_PR_CAS_O(S, T)						\
 	CK_CC_INLINE static bool					\
 	ck_pr_cas_##S##_value(T *target, T compare, T set, T *v)	\
 	{								\
-		set = __sync_val_compare_and_swap(target, compare, set);\
-		*v = set;						\
-		return (set == compare);				\
+		bool z;							\
+		z = __atomic_compare_exchange(target, &compare, &set,	\
```

Comment:
It's annoying that we need these changes, but last time I looked it was really awkward to get the sync builtins to work.